+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+                    obj.w.x(1:opts.N_stages,1:(opts.N_finite_elements(ii)-1),opts.n_s+rbp).ub = inf*ones(dims.n_x, 1);
+                end
+            end
+
+            % build z_alg, z_impulse, and z_alg_f_x variable groups
+            z_alg = {obj.w.lambda_normal, obj.w.y_gap};
+            z_impulse = {obj.w.Lambda_normal, obj.w.Y_gap, obj.w.P_vn, obj.w.N_vn};
+            z_alg_f_x = {obj.w.lambda_normal};
+            if model.friction_exists
+                z_alg = [z_alg, {obj.w.lambda_tangent}];
+                z_impulse = [z_impulse, {obj.w.Lambda_tangent}];
+                z_alg_f_x = [z_alg_f_x, {obj.w.lambda_tangent}];
+                switch opts.friction_model
+                  case 'Polyhedral'
+                    z_alg = [z_alg,{obj.w.gamma_d,obj.w.beta_d,obj.w.delta_d}];
+                    z_impulse = [z_impulse,{obj.w.Gamma_d,obj.w.Beta_d,obj.w.Delta_d}];
+                  case 'Conic'
+                    z_alg = [z_alg,{obj.w.gamma,obj.w.beta}];
+                    z_impulse = [z_impulse,{obj.w.Gamma,obj.w.Beta}];
+                    switch opts.conic_model_switch_handling
+                      case 'Plain'
+                        % no extra vars
+                      case 'Abs'
+                        z_alg = [z_alg,{obj.w.p_vt,obj.w.n_vt}];
+                        z_impulse = [z_impulse,{obj.w.P_vt,obj.w.N_vt}];
+                      case 'Lp'
+                        z_alg = [z_alg,{obj.w.p_vt,obj.w.n_vt,obj.w.alpha_vt}];
+                        z_impulse = [z_impulse,{obj.w.P_vt,obj.w.N_vt,obj.w.Alpha_vt}];
+                    end
+                end
+            end
+            if opts.lift_velocity_state
+                z_alg_f_x = [z_alg_f_x, {obj.w.z_v}]
+            end
+            obj.z_alg = vdx.VariableGroup(z_alg);
+            obj.z_impulse = vdx.VariableGroup(z_impulse);
+            obj.z_alg_f_x = vdx.VariableGroup(z_alg_f_x);
+        end
+
+        function generate_direct_transcription_constraints(obj)
+            import casadi.*
+            model = obj.model;
+            opts = obj.opts;
+            dcs = obj.dcs;
+            dims = obj.dcs.dims;
+
+            rbp = ~opts.right_boundary_point_explicit;
+
+            v_global = obj.w.v_global();
+            p_global = obj.p.p_global();
+
+            x_0 = obj.w.x(0,0,opts.n_s);
+            z_0 = obj.w.z(0,0,opts.n_s);
+            
+            obj.g.z(0,0,opts.n_s) = {dcs.g_z_fun(x_0, z_0, obj.w.u(1), v_global, [p_global;obj.p.p_time_var(1)])};
+            
+            x_prev = obj.w.x(0,0,opts.n_s);
+            for ii=1:opts.N_stages
+                h0 = obj.p.T().val/(opts.N_stages*opts.N_finite_elements(ii));
+                
+                ui = obj.w.u(ii);
+                p_stage = obj.p.p_time_var(ii);
+                p = [p_global;p_stage];
+                if obj.opts.use_speed_of_time_variables && opts.local_speed_of_time_variable
```

Comment:
having `problem_options.time_optimal_problem = 1` works (can be tested on `new_design_cls_discs.m`

having `use_speed_of_time_variables = 1` (and `local_speed_of_time_variable = 1`)  leads to some noninstructive errors.

Reply:
This error is fixed. Was a typo.